src/main/java/com/google/cloud/spanner/pgadapter/statements/IntermediateStatement.java

```diff
@@ -47,43 +51,23 @@
   private static final char SINGLE_QUOTE = '\'';
 
   public IntermediateStatement(String sql, Connection connection) throws SQLException {
-    this();
     this.sql = sql;
+    this.sqlWithoutComments = PARSER.removeCommentsAndTrim(sql);
     this.statements = parseStatements(sql);
-    this.command = parseCommand(sql);
+    this.command = parseCommand(sqlWithoutComments);
     this.connection = connection;
     this.statement = connection.createStatement();
-  }
-
-  protected IntermediateStatement() {
-    this.executed = false;
-    this.exception = null;
-    this.resultType = null;
-    this.hasMoreData = false;
-    this.statementResult = null;
-    this.updateCount = null;
-  }
-
-  /**
-   * Extracts what type of result exists within the statement. In JDBC a statement update count is
-   * positive if it is an update statement, 0 if there is no result, or negative if there are
```

Comment:
A zero update count does not mean that there is no result. It means that zero rows were updated/deleted.